### PR TITLE
Update .NET SDK to 6.0.100-preview.2.21155.3

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     }
   },
   "sdk": {
-    "version": "5.0.104",
+    "version": "6.0.100-preview.2.21155.3",
     "allowPrerelease": true,
     "rollForward": "latestMajor"
```
Comment (bricelam) [inline global.json:16]: Note, contributors will need to [install this version](https://dotnet.microsoft.com/download/dotnet/6.0) if they don't use our `startvs.cmd`

File: benchmark/EFCore.Benchmarks/EFCore.Benchmarks.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (bricelam) [inline benchmark/EFCore.Benchmarks/EFCore.Benchmarks.csproj:4]: Tests and benchmarks updated to 6.0; product remains at 5.0 (for now)

File: global.json (hunk 1 above)
Comment (smitpatel) [reply global.json:16]: Could we add this to https://github.com/dotnet/efcore/blob/main/docs/getting-and-building-the-code.md? (assuming installing this SDK and opening VS just works, earlier it didn't)

**Review by smitpatel — APPROVED** [review]

File: global.json (hunk 1 above)
Comment (roji) [reply global.json:16]: As long as it's preview SDKs (as opposed to dailies), that seems pretty reasonable to me... Yeah, a note in in the building docs seems appropriate...

**Comment by bricelam** [comment]
```
error NETSDK1148: Found multiple publish output files with the same relative path: test\EFCore.Design.Tests\xunit.runner.json, test\xunit.runner.json. [test\EFCore.Design.Tests\EFCore.Design.Tests.csproj]
```

**Comment by bricelam** [comment]
Stupid bot. 😠

**Comment by smitpatel** [comment]
ouch! The bots are taking over the world.

Seems like a new validation added in 6.0 https://github.com/dotnet/aspnetcore/issues/29524
If removing xunit works then we can go with that. Or we can unblock using work-around from the thread.

**Comment by smitpatel** [comment]
Internal build URL https://dev.azure.com/dnceng/public/_build/results?buildId=1040380&view=results

**Comment by bricelam** [comment]
xunit.runner.json wasn't used anymore (it was left over from .NET Framework)

**Comment by bricelam** [comment]
I'll keep an eye on the build to make sure it passes

**Comment by bricelam** [comment]
cc @Pilchie This is the second time today the bot has merged my PR without waiting for CI. Happens immediately after a force-push with additional changes.

**Comment by smitpatel** [comment]
I sent mail to bot team about this.

**Pull request merged**